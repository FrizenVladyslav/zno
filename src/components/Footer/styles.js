import { StyleSheet } from 'aphrodite/no-important'
import { responsive } from 'assets/styles/mixins'

export default StyleSheet.create({
  // @ts-ignore
  footer: {
    borderLeft: 'none',
    borderRight: 'none',
    marginTop: '100px !important',
    overflow: 'hidden !important',
    ':nth-child(1n) > .container': {
      display: 'flex',
      alignItems: 'flex-start',
      ':nth-child(1n) > div': {
        alignSelf: 'flex-end',
        ':nth-child(1n) > p': {
          whiteSpace: 'nowrap',
        },
      },
    },

    ':nth-child(1n) ul': {
      display: 'flex',
      flexDirection: 'column',
      flexWrap: 'wrap',
      minWidth: 200,
      maxHeight: 150,
      marginLeft: 200,
      paddingTop: 19,
      listStyleType: 'none',
      ':nth-child(1n) > li': {
        marginRight: 20,
        ':first-child': {
          fontWeight: 700,
          marginTop: -19,
        },
        ':nth-child(1n) > a': {
          color: '#333',
          ':hover': {
            color: '#333',
            textDecoration: 'underline',
          },
        },
      },
    },
    ...responsive(992, {
      ':nth-child(1n) ul': {
        marginLeft: 100,
      },
    }),
    ...responsive(768, {
      ':nth-child(1n) ul': {
        marginLeft: 50,
      },
    }),
    ...responsive(600, {
      ':nth-child(1n) > .container': {
        ':nth-child(1n) > div': {
          display: 'none',
        },

        ':nth-child(1n) > ul': {
          margin: 0,
          marginTop: -50,
          paddingLeft: 30,
          minWidth: 100,
          height: 'auto',
          flexWrap: 'nowrap',
        },
      },
    }),
  },
})
