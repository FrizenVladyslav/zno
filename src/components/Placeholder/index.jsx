import React from 'react'
import PropTypes from 'prop-types'
import { Placeholder } from 'semantic-ui-react'

const PlaceholderGenerator = ({ paragraphs }) => {
  return (
    <Placeholder>
      {paragraphs.map((line, index) => (
        <Placeholder.Paragraph key={index}>
          {[...Array(line)].map((l, index) => (
            <Placeholder.Line key={index} />
          ))}
        </Placeholder.Paragraph>
      ))}
    </Placeholder>
  )
}

PlaceholderGenerator.propTypes = {
  paragraphs: PropTypes.arrayOf(PropTypes.number),
}

export default PlaceholderGenerator
