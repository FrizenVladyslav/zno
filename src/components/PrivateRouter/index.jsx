import React from 'react'
import { Route, Redirect } from 'react-router'
import { toast } from 'react-toastify'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

const PrivateRoute = props => {
  let allow = false

  // here private logic ...
  allow = props.roles.includes(props.role)

  // show a notification
  if (!allow) toast.info(props.message || `You have not access to this page`)

  return allow ? (
    <Route {...props} />
  ) : (
    <Redirect to={props.redirectTo ? props.redirectTo : '/'} />
  )
}

PrivateRoute.propTypes = {
  roles: PropTypes.arrayOf(PropTypes.string).isRequired,
  message: PropTypes.string,
  redirectTo: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
}

// @ts-ignore
export default connect(({ user }) => ({ role: user.role }))(PrivateRoute)
