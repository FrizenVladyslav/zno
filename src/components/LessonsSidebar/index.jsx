import React from 'react'
import { Menu, Segment } from 'semantic-ui-react'
import { css } from 'aphrodite'

import styles from './styles'

const Sidebar = () => {
  return (
    <Menu vertical className={css(styles.sidebar)}>
      <Menu.Item header>Предмет</Menu.Item>
      <Menu.Item name="Українська мова" />
    </Menu>
  )
}

export default Sidebar
