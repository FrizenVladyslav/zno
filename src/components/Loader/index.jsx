import React from 'react'
import PropTypes from 'prop-types'
import { Loader as SemanticLoader } from 'semantic-ui-react'
import { css } from 'aphrodite/no-important'

import styles from './styles'

const Loader = props => {
  return (
    <div
      className={css(styles.loaderContainer)}
      style={{ minHeight: props.height }}
    >
      <SemanticLoader active inline="centered" />
    </div>
  )
}

Loader.propTypes = {
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
}

export default Loader
