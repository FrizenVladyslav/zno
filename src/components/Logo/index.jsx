import React from 'react'

import { Link } from 'react-router-dom'
import { Icon } from 'semantic-ui-react'
import { css } from 'aphrodite/no-important'
import styles from './styles'

const Logo = props => {
  return (
    <Link to="/" className={css(styles.logo)}>
      <Icon name="student" size="huge" />
      <h2>Краще за всіх</h2>
    </Link>
  )
}

export default Logo
