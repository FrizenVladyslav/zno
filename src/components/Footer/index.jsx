import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { Container, Segment } from 'semantic-ui-react'
import { css } from 'aphrodite/no-important'
import Logo from '../Logo'

import styles from './styles'

const Footer = props => {
  return (
    <Segment placeholder className={css(styles.footer)}>
      <Container>
        <div>
          <Logo style={{ alignSelf: 'flex-end' }} />
          <p>&copy; Всі права захищені)</p>
        </div>
        <ul>
          <li>Предмети:</li>
          {props.lessons &&
            props.lessons.map(lesson => (
              <li key={lesson._id}>
                <Link to={`/lessons/${lesson._id}`}>{lesson.title}</Link>
              </li>
            ))}
        </ul>
        <ul>
          <li>Інші посилання:</li>
          <li>
            <Link to="/login">Увійти</Link>
          </li>
          <li>
            <Link to="/lessons">Зайняття</Link>
          </li>
        </ul>
      </Container>
    </Segment>
  )
}

// @ts-ignore
export default connect(({ lesson }) => ({ lessons: lesson.lessons }))(Footer)
