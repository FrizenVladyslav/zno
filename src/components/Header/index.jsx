import React, { useState } from 'react'
import { connect } from 'react-redux'
import { Menu } from 'semantic-ui-react'
import { NavLink } from 'react-router-dom'
import { css } from 'aphrodite'
import Logo from '../Logo'
import ConfirmModal from '../Modals/ConfirmModal'

import * as userActions from 'actions/user'

import styles from './styles'

const Header = props => {
  const [modal, triggerModal] = useState(false)

  return (
    <>
      <Menu stackable className={css(styles.menu)}>
        <Menu.Header>
          <Logo />
        </Menu.Header>
        <Menu.Menu position="right">
          {props.user.role === 'admin' && (
            <Menu.Item name="admin" link as={NavLink} to="/admin">
              Адміністрування
            </Menu.Item>
          )}
          <Menu.Item name="lessons" link as={NavLink} to="/lessons">
            Предмети
          </Menu.Item>
          {!!props.user._id ? (
            <Menu.Item link onClick={() => triggerModal(true)}>
              {props.user.nick}
            </Menu.Item>
          ) : (
            <Menu.Item as={NavLink} link name="sign-in" to="/login">
              Увійти
            </Menu.Item>
          )}
        </Menu.Menu>
      </Menu>
      <ConfirmModal
        icon="sign-out"
        message="Ви дійсно бажаєте вийти?"
        open={modal}
        onClose={() => triggerModal(false)}
        onSuccess={userActions.signOut}
        title="Вийти"
      />
    </>
  )
}

// @ts-ignore
export default connect(({ user }) => ({ user }))(Header)
