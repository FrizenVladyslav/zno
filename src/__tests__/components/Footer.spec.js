// @ts-nocheck
import React from 'react'
import { shallow } from 'enzyme'
import { StyleSheetTestUtils } from 'aphrodite'
import { Footer } from 'components/Footer'

describe('footer', () => {
  afterEach(() => {
    return new Promise(resolve => {
      StyleSheetTestUtils.clearBufferAndResumeStyleInjection()
      return process.nextTick(resolve)
    })
  })

  let lessons = [
    {
      _id: '1',
      title: 'title1',
    },
    {
      _id: '2',
      title: 'title2',
    },
  ]
  const footer = shallow(<Footer lessons={lessons} />)

  it('renders properly', () => {
    expect(footer).toMatchSnapshot()
  })

  it('renders 2 lessons', () => {
    expect(
      footer
        .find('ul')
        .first()
        .children()
    ).toHaveLength(3)
  })

  describe('correct render first lesson link', () => {
    const link = footer
      .find('ul')
      .first()
      .find('Link')
      .at(0)

    it('title', () => {
      expect(link.props().children).toBe(lessons[0].title)
    })

    it('path', () => {
      expect(link.props().to).toBe(`/lessons/${lessons[0]._id}`)
    })
  })
})
