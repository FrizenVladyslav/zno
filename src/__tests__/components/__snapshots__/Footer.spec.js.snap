// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer
    lessons={
      Array [
        Object {
          "_id": "1",
          "title": "title1",
        },
        Object {
          "_id": "2",
          "title": "title2",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Container>
        <div>
          <Logo
            style={
              Object {
                "alignSelf": "flex-end",
              }
            }
          />
          <p>
            © Всі права захищені)
          </p>
        </div>
        <ul>
          <li>
            Предмети:
          </li>
          <li>
            <Link
              replace={false}
              to="/lessons/1"
            >
              title1
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/lessons/2"
            >
              title2
            </Link>
          </li>
        </ul>
        <ul>
          <li>
            Інші посилання:
          </li>
          <li>
            <Link
              replace={false}
              to="/login"
            >
              Увійти
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/lessons"
            >
              Зайняття
            </Link>
          </li>
        </ul>
      </Container>,
      "className": "footer_7e9spg",
      "placeholder": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div>
            <Logo
              style={
                Object {
                  "alignSelf": "flex-end",
                }
              }
            />
            <p>
              © Всі права захищені)
            </p>
          </div>,
          <ul>
            <li>
              Предмети:
            </li>
            <li>
              <Link
                replace={false}
                to="/lessons/1"
              >
                title1
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/lessons/2"
              >
                title2
              </Link>
            </li>
          </ul>,
          <ul>
            <li>
              Інші посилання:
            </li>
            <li>
              <Link
                replace={false}
                to="/login"
              >
                Увійти
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/lessons"
              >
                Зайняття
              </Link>
            </li>
          </ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Logo
                style={
                  Object {
                    "alignSelf": "flex-end",
                  }
                }
              />,
              <p>
                © Всі права захищені)
              </p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "style": Object {
                  "alignSelf": "flex-end",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "© Всі права захищені)",
              },
              "ref": null,
              "rendered": "© Всі права захищені)",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                Предмети:
              </li>,
              Array [
                <li>
                  <Link
                    replace={false}
                    to="/lessons/1"
                  >
                    title1
                  </Link>
                </li>,
                <li>
                  <Link
                    replace={false}
                    to="/lessons/2"
                  >
                    title2
                  </Link>
                </li>,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Предмети:",
              },
              "ref": null,
              "rendered": "Предмети:",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/lessons/1"
                >
                  title1
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "title1",
                  "replace": false,
                  "to": "/lessons/1",
                },
                "ref": null,
                "rendered": "title1",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/lessons/2"
                >
                  title2
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "title2",
                  "replace": false,
                  "to": "/lessons/2",
                },
                "ref": null,
                "rendered": "title2",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                Інші посилання:
              </li>,
              <li>
                <Link
                  replace={false}
                  to="/login"
                >
                  Увійти
                </Link>
              </li>,
              <li>
                <Link
                  replace={false}
                  to="/lessons"
                >
                  Зайняття
                </Link>
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Інші посилання:",
              },
              "ref": null,
              "rendered": "Інші посилання:",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/login"
                >
                  Увійти
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Увійти",
                  "replace": false,
                  "to": "/login",
                },
                "ref": null,
                "rendered": "Увійти",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/lessons"
                >
                  Зайняття
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Зайняття",
                  "replace": false,
                  "to": "/lessons",
                },
                "ref": null,
                "rendered": "Зайняття",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Container>
          <div>
            <Logo
              style={
                Object {
                  "alignSelf": "flex-end",
                }
              }
            />
            <p>
              © Всі права захищені)
            </p>
          </div>
          <ul>
            <li>
              Предмети:
            </li>
            <li>
              <Link
                replace={false}
                to="/lessons/1"
              >
                title1
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/lessons/2"
              >
                title2
              </Link>
            </li>
          </ul>
          <ul>
            <li>
              Інші посилання:
            </li>
            <li>
              <Link
                replace={false}
                to="/login"
              >
                Увійти
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/lessons"
              >
                Зайняття
              </Link>
            </li>
          </ul>
        </Container>,
        "className": "footer_7e9spg",
        "placeholder": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <div>
              <Logo
                style={
                  Object {
                    "alignSelf": "flex-end",
                  }
                }
              />
              <p>
                © Всі права захищені)
              </p>
            </div>,
            <ul>
              <li>
                Предмети:
              </li>
              <li>
                <Link
                  replace={false}
                  to="/lessons/1"
                >
                  title1
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/lessons/2"
                >
                  title2
                </Link>
              </li>
            </ul>,
            <ul>
              <li>
                Інші посилання:
              </li>
              <li>
                <Link
                  replace={false}
                  to="/login"
                >
                  Увійти
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/lessons"
                >
                  Зайняття
                </Link>
              </li>
            </ul>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Logo
                  style={
                    Object {
                      "alignSelf": "flex-end",
                    }
                  }
                />,
                <p>
                  © Всі права захищені)
                </p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "style": Object {
                    "alignSelf": "flex-end",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "© Всі права захищені)",
                },
                "ref": null,
                "rendered": "© Всі права захищені)",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  Предмети:
                </li>,
                Array [
                  <li>
                    <Link
                      replace={false}
                      to="/lessons/1"
                    >
                      title1
                    </Link>
                  </li>,
                  <li>
                    <Link
                      replace={false}
                      to="/lessons/2"
                    >
                      title2
                    </Link>
                  </li>,
                ],
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Предмети:",
                },
                "ref": null,
                "rendered": "Предмети:",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/lessons/1"
                  >
                    title1
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "title1",
                    "replace": false,
                    "to": "/lessons/1",
                  },
                  "ref": null,
                  "rendered": "title1",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/lessons/2"
                  >
                    title2
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "title2",
                    "replace": false,
                    "to": "/lessons/2",
                  },
                  "ref": null,
                  "rendered": "title2",
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  Інші посилання:
                </li>,
                <li>
                  <Link
                    replace={false}
                    to="/login"
                  >
                    Увійти
                  </Link>
                </li>,
                <li>
                  <Link
                    replace={false}
                    to="/lessons"
                  >
                    Зайняття
                  </Link>
                </li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Інші посилання:",
                },
                "ref": null,
                "rendered": "Інші посилання:",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/login"
                  >
                    Увійти
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Увійти",
                    "replace": false,
                    "to": "/login",
                  },
                  "ref": null,
                  "rendered": "Увійти",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/lessons"
                  >
                    Зайняття
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Зайняття",
                    "replace": false,
                    "to": "/lessons",
                  },
                  "ref": null,
                  "rendered": "Зайняття",
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
